http-webhook.md

```diff
@@ -0,0 +1,376 @@
+containing the access token will be logged, it SHOULD NOT be used unless it is
+impossible to transport the access token in the "Authorization" request header
+field or the HTTP request entity-body. All further caveats cited in
+[RFC6750][RFC6750] apply equivalently.
+
+## 4. Abuse Protection
+
+Any system that allows registration of and delivery of notifications to
+arbitary HTTP endpoints can potentially be abused such that someone maliciously
+or inadvertently registers the address of a system that does not expect such
+requests and for which the registering party is not authorized to perform such
+a registration. In extreme cases, a notification infrastructure could be abused
+to launch denial-of-service attacks against an arbitrary web-site.
+
+To protect the sender from being abused in such a way, a legitimate delivery
+target needs to indicate that it agrees with notifications being delivered to it.
+
+Reaching the delivery agreement is realized using the following validation
+handshake. The handshake can either be executed immediately at registration
+time or as a "pre-flight" request immediately preceding a delivery. If the
+handshake outcome is that the delivery is denied, the sender MUST NOT deliver
+events to the target.
+
+It is important to understand is that the handshake does not aim to establish
+an authentication or authorization context. It only serves to protect the
+sender from being told to a push to a destination that is not expecting the
+traffic. While this specification mandates use of an authorization model, this
+mandate is not sufficient to protect any arbitrary website from unwanted
+traffic if that website doesn't implement access control and therefore ignores
+the `Authorization` header.
+
+Delivery targets SHOULD support the abuse protection feature. If a target does
+not support the feature, the sender MAY choose not to send to the target, at
+all, or send only at a very low request rate.
+
+### 4.1. Validation request
+
+The validation request uses the HTTP [OPTIONS][OPTIONS] method. The request
+is directed to the exact resource target URI that is being registered.
+
+With the validation request, the sender asks the target for permission to
+send notifications, and it can declare a desired request rate (requests per
+minute).
+
+The delivery target will respond with a permission statement and the permitted
+request rate.
+
+The following header fields are for inclusion in the validation request.
+
+#### 4.1.2. WebHook-Request-Origin
+
+The `WebHook-Request-Origin` header MUST be included in the validation request
+and requests permission to send notifications from this sender, and contains a
+DNS expression that identifies the sending system, for example
+"eventemitter.example.com". The value is meant to summarily identify all sender
+instances that act on the behalf of a certain system, and not an individual
+host.
+
+After the handshake and if permission has been granted, the sender MUST use the
```

Comment:
Why use `WebHook-Request-Origin` only for abuse protection handshake request? Would it make sense to use `Origin` both for the handshake and later on delivery? The two requests are already different enough, by HTTP method - handshake uses OPTIONS and delivery uses POST.

Reply:
I don't understand what the 'Origin' header would add for the handshake itself. It is mandated for the delivery request.

Reply:
Currently it's written that:
```
The `WebHook-Request-Origin` header MUST be included in the validation request
and requests permission to send notifications from this sender, and contains a
DNS expression that identifies the sending system, for example
"eventemitter.example.com"

After the handshake and if permission has been granted, the sender MUST use the
`Origin` request header for each delivery request, with the value matching that
of this header.
```

If I understood correctly, in handshake/OPTIONS request event emitter has to send `WebHook-Request-Origin` header with value equal to the hostname of `Origin` header that it will be sending in delivery/POST requests.

Dilemma I have is why not send `Origin` during handshake with same value that will be used later in delivery, why new custom header when standard `Origin` header will be sent anyway during handshake too and probably with same value as during delivery and from it hostname can be extracted if just hostname is needed?